addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -510,7 +510,10 @@ odoo.define('pos_coupon.pos', function (require) {
         _calculateRewards: async function () {
             const rewardsContainer = new RewardsContainer();
 
-            if (this._getRegularOrderlines().length === 0) {
+            const refundOrderlines = _order_super.get_orderlines
+                .apply(this, arguments)
+                .filter((line) => line.refunded_orderline_id);
```

Comment:
We should modify the function '_getRegularOrderlines' instead of modifying everyplaces where it is called.

And modify:
  get_orderlines: function () {
            return [...this._getRegularOrderlines(), ...this._getRewardLines()];
        },
  To keep the refunded lines, maybe just calling super, and sort them to have the rewarlines at the end.

Reply:
Wow, it worked like a charm! :)